database-commons/src/main/java/io/cdap/plugin/db/sink/ETLDBOutputFormat.java

```diff
@@ -67,14 +70,27 @@ public RecordWriter<K, V> getRecordWriter(TaskAttemptContext context) throws IOE
     String tableName = dbConf.getOutputTableName();
     String[] fieldNames = dbConf.getOutputFieldNames();
     final int batchSize = conf.getInt(COMMIT_BATCH_SIZE, DEFAULT_COMMIT_BATCH_SIZE);
+    final String operationName = conf.get(OPERATION_NAME);
+    String relationTableKey = null;
+    String[] listKeys = new String[0];
+    if (conf.get(RELATION_TABLE_KEY) != null) {
+      relationTableKey = conf.get(RELATION_TABLE_KEY);
+      listKeys = relationTableKey.split(",");
+    }
 
     if (fieldNames == null) {
       fieldNames = new String[dbConf.getOutputFieldCount()];
     }
 
     try {
       Connection connection = getConnection(conf);
-      PreparedStatement statement = connection.prepareStatement(constructQuery(tableName, fieldNames));
+      PreparedStatement statement;
+      if (relationTableKey != null) {
```

Comment:
check for listkeys instead, relationTableKey can be used inside the if block only